x-pack/plugins/watcher/server/routes/api/indices/register_get_route.ts

```diff
@@ -26,44 +27,49 @@ function getIndexNamesFromAliasesResponse(json: Record<string, any>) {
   );
 }
 
-function getIndices(dataClient: ILegacyScopedClusterClient, pattern: string, limit = 10) {
-  return dataClient
-    .callAsCurrentUser('indices.getAlias', {
+async function getIndices(dataClient: IScopedClusterClient, pattern: string, limit = 10) {
+  const aliasResult = await dataClient.asCurrentUser.indices.getAlias(
+    {
       index: pattern,
+    },
+    {
       ignore: [404],
-    })
-    .then((aliasResult: any) => {
-      if (aliasResult.status !== 404) {
-        const indicesFromAliasResponse = getIndexNamesFromAliasesResponse(aliasResult);
-        return indicesFromAliasResponse.slice(0, limit);
-      }
+    }
+  );
 
-      const params = {
-        index: pattern,
-        ignore: [404],
-        body: {
-          size: 0, // no hits
-          aggs: {
-            indices: {
-              terms: {
-                field: '_index',
-                size: limit,
-              },
+  if (aliasResult.statusCode !== 404) {
+    const indicesFromAliasResponse = getIndexNamesFromAliasesResponse(aliasResult.body);
+    return indicesFromAliasResponse.slice(0, limit);
+  }
+
+  const response = await dataClient.asCurrentUser.search(
+    {
+      index: pattern,
+      body: {
+        size: 0, // no hits
+        aggs: {
+          indices: {
+            terms: {
+              field: '_index',
+              size: limit,
             },
           },
         },
-      };
+      },
+    },
+    {
+      ignore: [404],
+    }
+  );
+  if (response.statusCode === 404 || !response.body.aggregations) {
+    return [];
+  }
+  const indices = response.body.aggregations.indices as MultiBucketAggregate<{ key: unknown }>;
 
-      return dataClient.callAsCurrentUser('search', params).then((response: any) => {
-        if (response.status === 404 || !response.aggregations) {
-          return [];
-        }
-        return response.aggregations.indices.buckets.map((bucket: any) => bucket.key);
-      });
-    });
+  return indices.buckets ? indices.buckets.map((bucket) => bucket.key) : [];
```

Comment:
I was going to suggest being even stricter here:

```js
return Array.isArray(indices.buckets) ? indices.buckets.map((bucket) => bucket.key) : [];
```

I'm imagining the response containing something unexpected, either because of a deliberate API change or because of a bug, and generating a 500 because `map` is undefined. But then I decided that it will be easier to debug a user complaining about a 500 and the accompanying stack trace in the logs, than it would be to debug a user complaining about silently disappearing information. So I think this is good as-is. :)